src/server/metadata/server/api_server.go

```diff
@@ -23,6 +27,65 @@ func NewMetadataServer(env Env) *APIServer {
 	}
 }
 
+// EditMetadata transactionally mutates metadata.  All operations are attempted, in order, but if
+// any fail, the entire operation fails.
 func (s *APIServer) EditMetadata(ctx context.Context, req *metadata.EditMetadataRequest) (*metadata.EditMetadataResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	res := &metadata.EditMetadataResponse{}
+	if err := dbutil.WithTx(ctx, s.env.DB, func(ctx context.Context, tx *pachsql.Tx) error {
```

Comment:
It might be worth promoting this logic — stripped of its gRPC RPC-handler details — up to `src/server/metadata`.

Reply:
Done.